docs/ko/reference/array/fill.md

```diff
@@ -5,24 +5,26 @@
 이 함수는 원본 배열을 변경하고, 제공된 값으로 시작 인덱스부터 끝 인덱스까지의 요소를 대체해요.
 시작 또는 끝 인덱스를 제공하지 않으면 배열 전체를 채워요.
 
+음수 인덱스를 사용할 수도 있어요. 이 경우 배열의 끝에서부터 인덱스를 계산해요.
+
 ## 인터페이스
 
 ```typescript
-function fill<T>(array: unknown[], value: T): T[];
-function fill<T, P>(array: T[], value: P, start: number): Array<T | P>;
-function fill<T, P>(array: T[], value: P, start: number, end: number): Array<T | P>;
+function fill<T, U>(arr: Array<T | U>, value: U): Array<T | U>;
```

Comment:
I was wondering why we use two type parameters here; Doesn't the original type signature already infer the the correctly?

```tsx
function fill<T>(array: unknown[], value: T): T[];
```

Reply:
Hello, thanks your review!

AS_IS
`fill<T>(array: unknown[], value: T): T[];` 

TO_BE
`function fill<T, U>(arr: Array<T | U>, value: U): Array<T | U>;` 

Flexibility: To separate the element type (T) and the value type (U) of the array, allowing the array to hold elements of different types.

Clarity: To clearly define the element type and the value type of the array, better expressing the intent of the code.
Replace unknown: There was also an intention to replace the unknown type.

However, based on your feedback, since the original simple type signature is sufficient for correct type inference, it seems better to define the type as `fill<T>(array: unknown[], value: T): T[];`

Shall we change the type definition back to the original?

Reply:
Yes, I guess we should change the definition to the original one, since behavior is identical and the original implementation is simple.

Reply:
I agree with your opinion. I have updated the original definition and revised the document accordingly. Thank you.

Original definition changed, and the document has been revised.

Reply:
I agree with your opinion. I have updated the original definition and revised the document accordingly. Thank you.

Original definition changed, and the document has been revised.